### ISSUE TITLE
memory errors running on server

### ISSUE THREAD

**Issue by alejandrojcastaneira** [flairNLP/flair#1941]
Hello,

I have an API served with NER models trained with Flair embeddings, the original version not the pooled one. However, I have noticed that the memory on the gpu might not be released correctly, after making some inferences, the memory starts to grow and after several thousand inferences are made, CUDA memory errors start to show, after the first one occurs all subsequent inferences throw this.

I have the setting embedding_storage_mode = "none" when making the predictions and have tried the other settings, but we observed the same behaviour.

I am currently restarting the server daily or when the Cuda error occurs, so sorry to comment on this, for our use case the models are performing great! and the predictions are excellent! 

Thank you very much for the great work!

Best regards

**Environment**
 - Linux 18.10:
 - flair-0.6.1:

**Comment by stefan-it** [comment]
I've seen some previous memory issues in Transformers library because of memory leaks in PyTorch. Could you specify what PyTorch version you're using or could you update to a more recent one :thinking:

**Comment by alejandrojcastaneira** [comment]
I'm using Pytorch=1.6.0 also with cuda==10.2, I will update now to the version 1.7.0 and I will let you know if I notice a change.

**Comment by alejandrojcastaneira** [comment]
Hello, I have observed similar behavior with Pytorch = 1.7.0, its memory consumption keeps increasing over time, although I agree that this may be related to Pytorch. 

I also observed that extremely long sentences also cause cuda errors. Will it be possible to use `AMP` also after the model is trained to make predictions? not only during training language models or other components. This could alleviate the gpu load when processing multiple concurrent requests.
